golang/cosmos/x/swingset/keeper/store_snapshotter.go

```diff
@@ -0,0 +1,601 @@
+package keeper
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/vm"
+	"github.com/Agoric/agoric-sdk/golang/cosmos/x/swingset/types"
+	vstoragetypes "github.com/Agoric/agoric-sdk/golang/cosmos/x/vstorage/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// ExportManifestFilename is the manifest filename which must be synchronized with the JS export/import tooling
+const ExportManifestFilename = "export-manifest.json"
+
+// UntrustedExportDataArtifactName is a special artifact name that the provider and consumer of an export can
+// use to indicate the presence of a synthetic artifact containing untrusted "export data". This artifact must
+// not end up in the list of artifacts imported by the JS import tooling (which would fail).
+const UntrustedExportDataArtifactName = "UNTRUSTED-EXPORT-DATA"
+
+const exportDataFilename = "export-data.jsonl"
+const untrustedExportDataFilename = "untrusted-export-data.jsonl"
```

Comment:
A note in case it doesn't come up later: untrusted export data is emitted as a sequence of line-feed-*terminated* JSON arrays, but https://jsonlines.org/ documents that lines are _separated_ and the validator at https://jsonlines.org/validator/ rejects input in which the last character is a line feed. The difference is subtle, but probably worth explicitly mentioning in e.g. golang/cosmos/x/swingset/keeper/store_snapshotter.go `RestoreSnapshot`.

Reply:
ah yeah, this is line terminated json, which is fine in all our parsers, both in JS and golang. I'll make a note.